server.js

```diff
@@ -22,23 +22,45 @@ io.on('connection', (socket) => {
     console.log('Client Connected')
 
     socket.on('chat', (message) => {
-        const room = Object.keys(socket.rooms)
-            .filter(x => x !== socket.id)
-
-        io.to(room[0]).emit('chat', message)
+        const room = filterRooms(socket)
+        notifyRoom(room, 'chat', message)
         console.log(message)
     })
 
     socket.on('private', (sender, message) => {
         // private message received
     })
-
-    socket.on('disconnect', () => {
-        io.emit('client disconnected')
+    
+    socket.on('leave-room', (name) => {
+        const room = filterRooms(socket)
+        const message = name + ' has disconnected'
```

Comment:
For future reference, you could use string literals ex.

    const message = `${name} has disconnected`